cms/articles/models.py

```diff
@@ -69,12 +71,21 @@ def latest_release(self, request: "HttpRequest") -> "TemplateResponse":
 
     @path("previous-releases/")
     def previous_releases(self, request: "HttpRequest") -> "TemplateResponse":
-        response: TemplateResponse = self.render(
-            request,
-            # TODO: update to include drafts when looking at previews holistically.
-            context_overrides={"pages": StatisticalArticlePage.objects.live().child_of(self).order_by("-release_date")},
-            template="templates/pages/statistical_article_page--previous-releases.html",
-        )
+        children = StatisticalArticlePage.objects.live().child_of(self).order_by("-release_date")
+        paginator = Paginator(children, per_page= settings.PREVIOUS_RELEASES_PER_PAGE)
+
+        try:
+            pages = paginator.page(int(request.GET.get("page", 1)))
+            ons_pagination_url_list = [{"url": "?page=" + str(n + 1)} for n in range(paginator.num_pages)]
```

Comment:
```suggestion
            ons_pagination_url_list = [{"url": f"?page={n}"} for n paginator.page_range]
```

See https://docs.djangoproject.com/en/5.1/topics/pagination/


or, if you prefer using range, then 
```suggestion
            ons_pagination_url_list = [{"url": f"?page={n}"} for n in range(1, paginator.num_pages + 1)]
```

as per https://docs.python.org/3.12/library/stdtypes.html#range

---

Kudos for the approach to satisfy the page urls list requirement for [the pagination component](https://service-manual.ons.gov.uk/design-system/components/pagination#example.example-pagination.nunjucks-macro-options)

Reply:
I liked your first suggestion but it needed an "in"

Reply:
overzealous deletion of characters. edited